### PR TITLE
Auth context sample

### PR DESCRIPTION
**Pull request by derisen**
## Purpose

* This PR adds a sample demonstrating how to enable auth context and handle claims challenges

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd ms-identity-javascript-react-tutorial
git checkout auth-context-sample
cd 6-AdvancedScenarios/3-call-api-acrs/API
npm install
cd ../SPA
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Review by kalyankrishna1 — CHANGES_REQUESTED** [review]
:clock1:

File: 6-AdvancedScenarios/3-call-api-acrs/README.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+1. In the app's registration screen, find and note the **Application (client) ID**. You use this value in your app's configuration file(s) later in your code.
+1. In the app's registration screen, select **Authentication** in the menu.
+   - If you don't have a platform added, select **Add a platform** and select the **Web** option.
+   - In the **Redirect URIs** section, enter the following redirect URIs.
+      - `http://localhost:5000/admin/redirect`
+1. Select **Save** to save your changes.
+1. In the app's registration screen, select the **Certificates & secrets** blade in the left to open the page where we can generate secrets and upload certificates.
+1. In the **Client secrets** section, select **New client secret**:
+   - Type a key description (for instance `app secret`),
+   - Select one of the available key durations (**In 1 year**, **In 2 years**, or **Never Expires**) as per your security posture.
```
Comment (kalyankrishna1) [inline 6-AdvancedScenarios/3-call-api-acrs/README.md:141]: >Select one of the available key durations (**In 1 year**, **In 2 years**, or **Never Expires**) [](http://example.com/codeflow?start=5&length=95)

maybe update the readme

File: 6-AdvancedScenarios/3-call-api-acrs/API/app.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
+const express = require('express');
+const session = require('express-session');
+const methodOverride = require('method-override');
+const cors = require('cors');
+const path = require('path');
+
+const msalWrapper = require('../../../../msal-express-wrapper/dist/index');
```
Comment (kalyankrishna1) [inline 6-AdvancedScenarios/3-call-api-acrs/API/app.js:7]: fix this

File: 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/sample.json [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    ],
+    "CodeConfiguration": [
+        {
+            "App": "service",
+            "SettingKind": "Replace",
+            "SettingFile": "\\..\\API\\authConfig.js",
+            "Mappings": [
+                {
+                    "key": "Enter_the_Application_Id_Here",
+                    "value": "client.AppId"
```
Comment (kalyankrishna1) [inline 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/sample.json:53]: >"value": "client.AppId" [](http://example.com/codeflow?start=20&length=23)

fix this

File: 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/sample.json [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    },
+    "AADApps": [
+        {
+            "Id": "service",
+            "Name": "msal-node-api-acrs",
+            "Kind": "WebApi",
+            "Audience": "AzureADMyOrg",
+            "HomePage": "http://localhost:5000/admin/home",
+            "ReplyUrls": "http://localhost:5000/admin/redirect",
+            "PasswordCredentials": "Auto",
```
Comment (kalyankrishna1) [inline 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/sample.json:18]: >PasswordCredentials [](http://example.com/codeflow?start=13&length=19)

Delegated permissions of Policy.Read.ConditionalAccess, Policy.ReadWrite.ConditionalAccess

File: 6-AdvancedScenarios/3-call-api-acrs/README.md [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+
+![Overview](./ReadmeFiles/Create-Fetch_Click.png)
+
+Select an operation in the web API and an auth context value to apply. Then select **Save or Update**. This updates this mapping in the local app's database. We advise you use the same auth context value for operations if possible, as this ensures that the user is redirected to Azure AD just once to perform the step-up authN.
+
+> :warning: When changing auth context mappings, have the user sign-out and sign-back in for the changes to take effect.
+
+1. Go to **Details** page to get details of data saved on the web API side in its database. You can select **Delete** if you need to delete a mapping from the local database.
+
+![Overview](./ReadmeFiles/ViewDetails.png)
```
Comment (kalyankrishna1) [inline 6-AdvancedScenarios/3-call-api-acrs/README.md:261]: replace the images

File: 6-AdvancedScenarios/3-call-api-acrs/README.md [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+The web API is now ready to challenge users for step-up auth for the selected operations.
+
+### Configure a conditional access policy to use auth context in Azure portal
+
+1. Navigate to **Azure Active Directory** > **Security** > **Conditional Access**
+1. Select **New policy** and go to **Cloud apps or actions**. In dropdown select **Authentication context**. The newly created auth context values will be listed for you to be used in this CA policy.
+
+![Overview](./ReadmeFiles/AuthContext.png)
+
+Select the value and create the policy as required. For example, you might want the user to satisfy a **MFA** challenge if the auth context value is **c1**.
```
Comment (kalyankrishna1) [inline 6-AdvancedScenarios/3-call-api-acrs/README.md:272]: >c1 [](http://example.com/codeflow?start=151&length=2)

Avoid using c1 and c2, instead, us the auth context name instead

File: 6-AdvancedScenarios/3-call-api-acrs/README.md (hunk 1 above)
Comment (derisen) [reply 6-AdvancedScenarios/3-call-api-acrs/README.md:141]: fixed!

File: 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/sample.json (hunk 3 above)
Comment (derisen) [reply 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/sample.json:53]: fixed!

File: 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/sample.json (hunk 4 above)
Comment (derisen) [reply 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/sample.json:18]: Added!

File: 6-AdvancedScenarios/3-call-api-acrs/README.md (hunk 5 above)
Comment (derisen) [reply 6-AdvancedScenarios/3-call-api-acrs/README.md:261]: will do

File: 6-AdvancedScenarios/3-call-api-acrs/README.md (hunk 6 above)
Comment (derisen) [reply 6-AdvancedScenarios/3-call-api-acrs/README.md:272]: changed!

File: 6-AdvancedScenarios/3-call-api-acrs/README.md (hunk 5 above)
Comment (derisen) [reply 6-AdvancedScenarios/3-call-api-acrs/README.md:264]: done!

**Pull request merged**